packages/@okta/vuepress-site/docs/guides/sign-in-to-spa-embedded-widget/main/react/basic-sign-in.md

```diff
@@ -0,0 +1,202 @@
+  const onSuccess = (tokens) => {
+    oktaAuth.handleLoginRedirect(tokens);
+  };
+
+  const onError = (err) => {
+    console.log('Sign in error:', err);
+  };
+
+  if (!authState) {
+    return <div>Loading ... </div>;
+  }
+
+  return authState.isAuthenticated ?
+    <Redirect to={{ pathname: '/' }}/> :
+    <OktaSignInWidget config={config} onSuccess={onSuccess} onError={onError}/>;
+};
+export default Login;
+```
+
+#### Callback route
+
+The [Okta React SDK](https://github.com/okta/okta-react) provides the [LoginCallback](https://github.com/okta/okta-react#logincallback) component for the callback route. It handles token parsing, token storage, and redirects the user to the `/` path. If a `SecureRoute` triggered the redirect, then the callback is directed to the secured route. See how the callback route component is called from the route definition file (`src/App.jsx`) in the [Connect the routes](#connect-the-routes) section.
+
+### Connect the routes
+
+Update `src/App.jsx` to include your project components and routes. The `Security` component controls the authentication flows, so it requires your OpenID Connect configuration. By default, `@okta/okta-react` redirects to Okta's sign-in page when the user isn't authenticated. In this example, `onAuthRequired` is overridden to redirect to the custom sign-in route instead.
+
+> **Note:** This example uses the `react-router-dom` version 5.x module for the `useHistory` and `Route` objects.
+
+```js
+import React from 'react';
+import { Route, useHistory } from 'react-router-dom';
+import { Security, SecureRoute, LoginCallback } from '@okta/okta-react';
+import { OktaAuth, toRelativeUrl } from '@okta/okta-auth-js';
+import Home from './Home';
+import Login from './Login';
+import Protected from './Protected';
+import Profile from './Profile';
+import config from './config';
+
+import logo from './logo.svg';
+import './App.css';
+
+const oktaAuth = new OktaAuth(config.oidc);
+
+const App = () => {
+  const history = useHistory();
+
+  const customAuthHandler = () => {
+    history.push('/login');
+  };
+
+  const restoreOriginalUri = async (_oktaAuth, originalUri) => {
+    history.replace(toRelativeUrl(originalUri || '', window.location.origin));
+  };
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>My Okta-React app</p>
+        <img src={logo} className="App-logo" alt="logo" />
+      <Security
+        oktaAuth={oktaAuth}
+        onAuthRequired={customAuthHandler}
+        restoreOriginalUri={restoreOriginalUri}
+      >
+        <Route path="/" exact component={Home} />
+        <SecureRoute path="/protected" component={Protected} />
+        <Route path="/login" render={() => <Login />} />
+        <Route path="/login/callback" component={LoginCallback} />
+    </Security>
+    </header>
+    </div>
+  );
+};
+
+export default App;
+
+```
+
+### Start your app
+
+To run and test your app, execute:
+
+```js
+npm start
```

Comment:
There are 70 warnings in my console when I run this command. Nothing is rendered in my browser. Here are the last two:

```
WARNING in ./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].oneOf[5].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[1].oneOf[5].use[2]!./node_modules/source-map-loader/dist/cjs.js!./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css)
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/root/okta/okta-signin-widget/target/sass/widgets/_jquery.qtip.scss' file: Error: ENOENT: no such file or directory, open '/root/okta/okta-signin-widget/target/sass/widgets/_jquery.qtip.scss'
 @ ./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css 8:6-261 22:17-24 26:7-21 58:25-39 59:36-47 59:50-64 63:6-73:7 64:54-65 64:68-82 70:42-53 70:56-70 72:21-28 83:0-231 83:0-231 84:22-29 84:33-47 84:50-64 61:4-74:5
 @ ./src/OktaSignInWidget.jsx 8:0-64
 @ ./src/Login.jsx 8:0-50 49:34-50
 @ ./src/App.js 11:0-28 71:45-50
 @ ./src/index.js 7:0-24 11:33-36

WARNING in ./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].oneOf[5].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[1].oneOf[5].use[2]!./node_modules/source-map-loader/dist/cjs.js!./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css)
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/root/okta/okta-signin-widget/target/sass/widgets/_mega-drop-down.scss' file: Error: ENOENT: no such file or directory, open '/root/okta/okta-signin-widget/target/sass/widgets/_mega-drop-down.scss'
 @ ./node_modules/@okta/okta-signin-widget/dist/css/okta-sign-in.min.css 8:6-261 22:17-24 26:7-21 58:25-39 59:36-47 59:50-64 63:6-73:7 64:54-65 64:68-82 70:42-53 70:56-70 72:21-28 83:0-231 83:0-231 84:22-29 84:33-47 84:50-64 61:4-74:5
 @ ./src/OktaSignInWidget.jsx 8:0-64
 @ ./src/Login.jsx 8:0-50 49:34-50
 @ ./src/App.js 11:0-28 71:45-50
 @ ./src/index.js 7:0-24 11:33-36

70 warnings have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.69.1 compiled with 70 warnings in 2727 ms
```